### PR TITLE
roachtest/grafana: add SQL metrics to CDC dashboard

### PR DESCRIPTION
**Pull request by andyyang890**
This patch adds a row to the CDC roachtest grafana dashboard with a few
SQL metrics (SQL Statements, SQL Byte Traffic, Service Latency) that
are of interest while doing performance testing. This addition is
valuable because the TestEng Grafana instance currently only works with
GCE roachtests, and there are times when we need to run tests on other
cloud platforms like AWS.

Informs #124432

Release note: None

### CODE REVIEW

**Comment by andyyang890** [comment]
Example screenshot:
Screenshot 2024-08-23 at 6 41 57 PM

**Review by rharding6373 — APPROVED** [review]
LGTM from a CDC perspective, though you should wait for a review from folks more familiar with grafana dashboards.

**Comment by DarrylWong** [comment]
Hmm, we already have these metrics under the SQL dashboard here: `pkg/cmd/roachprod/grafana/configs/sql.json`. Unfortunately they live under the `roachprod/grafana` package so it's a bit cumbersome to access through a roachtest since those aren't embedded. 

Did a quick refactor to move everything under `roachtest/grafana` to `roachprod/grafana` and it seems to work by just adding `WithGrafanaDashboardJSON(grafana.SQLGrafana)`. Not sure why we would want to distinguish between roachprod and roachtest dashboards. Maybe @herkolategan would know since he did the initial refactor to make the dashboards embedded?

**Comment by DarrylWong** [comment]
Ah nevermind, figured it out. Andy pointed out to me that some of stats take in a `Cluster` variable which isn't set up for the internal grafana instance. Probably not too hard to teach it to do that, but this is probably a fine change for now.

**Comment by herkolategan** [comment]
@DarrylWong Good point, we should probably consolidate these, I'll create an issue for it. But for now this change is all good.

**Review by herkolategan — APPROVED** [review]

**Review by DarrylWong — APPROVED** [review]

**Pull request merged**
